compile.py

```diff
@@ -0,0 +1,608 @@
+#!/usr/bin/env python3
+
+import ast
+import json
+import warnings
+import outlines
+
+INDENT = 2
+
+class CantCompile(Exception):
+    def __init__(self, tree, hint=None):
+        super().__init__(f"Could not compile `{ast.unparse(tree)}`")
+        self.tree = tree
+        self.hint = hint
+        self.msg = None
+
+ISSUES = []
+
+def catch_issues(f):
+    def wrapped(tree, *args, **kwargs):
+        try:
+            return f(tree, *args, **kwargs)
+        except AssertionError as e2:
+            try:
+                return find_hint(tree, "js")
+            except CantCompile as e:
+                e.msg = str(e2)
+                ISSUES.append(e)
+                return "/* " + ast.unparse(tree) + " */"
+        except CantCompile as e:
+            if not e.hint:
+                try:
+                    return find_hint(tree, "js")
+                except CantCompile as e2:
+                    e = e2
+            ISSUES.append(e)
+            return "/* " + ast.unparse(e.tree) + " */"
+    return wrapped
+
+HINTS = []
+
+def read_hints(f):
```

Comment:
Ah, I should. The real answer is—if the compiler fails to compile, it will print the hints you need to add with a `???` for what you have to fill in. You copy that into the relevant hints file and it'll do its thing.